### ISSUE TITLE
Allow envvars in the `func.yaml` config.

### ISSUE THREAD

**Issue by matejvasek** [boson-project/faas#234]
I think it might be useful to be able to use envvars in `func.yaml`

Sample:
```yaml
envVars:
  SOME_SECRET: {{env.API_KEY}}
```

**Comment by zroubalik** [comment]
@matejvasek this could be closed, right?

**Comment by matejvasek** [comment]
@zroubalik actually no, this is not about setting evvars to ksvc, but about using in config.

**Comment by matejvasek** [comment]
e.g. in java property file you can write
```
my.prop=${env:JAVA_HOME}
```

**Comment by matejvasek** [comment]
So you can fill envvar from envvar.

**Comment by matejvasek** [comment]
@zroubalik Does this make sense?

**Comment by zroubalik** [comment]
@matejvasek gotcha :)

**Comment by zroubalik** [comment]
Referencing a Kubernetes Secret would be nice to have to

**Comment by matejvasek** [comment]
> Referencing a Kubernetes Secret would be nice to have too

Absolutely, but it will decrease level of abstraction from k8s.

**Comment by zroubalik** [comment]
> > Referencing a Kubernetes Secret would be nice to have too
> 
> Absolutely, but it will decrease level of abstraction from k8s.

Yeah, but Secrets are fundamental building blocks heavily used even by non k8s experienced users. Many times, Secrets with creds are preconfigured by admins or ops and devs are consuming them.

**Comment by matejvasek** [comment]
Ian just ask me about such a feature like this.

**Comment by matejvasek** [comment]
@boson-project/core what syntax do we prefer `${env.NAME}` or `{{evn.NAME}}` or something else?

**Comment by lance** [comment]
@matejvasek I don't personally have a preference. Whatever is easiest to code a parser for I guess. :smile:
